src/main/kotlin/com/sourcegraph/cody/agent/protocol/Speaker.kt

```diff
@@ -1,12 +1,13 @@
 package com.sourcegraph.cody.agent.protocol
 
-import com.google.gson.JsonPrimitive
-import com.google.gson.JsonSerializer
+import com.google.gson.JsonDeserializer
 import com.google.gson.annotations.SerializedName
 
 enum class Speaker(val speaker: String) {
   @SerializedName("human") HUMAN("human"),
   @SerializedName("assistant") ASSISTANT("assistant")
 }
 
-val SpeakerSerializer = JsonSerializer<Speaker> { src, _, _ -> JsonPrimitive(src.speaker) }
+val speakerDeserializer = JsonDeserializer { src, _, _ ->
+  Speaker.values().find { it.speaker == src.asString }
```

Comment:
still, not the best but rather simple and working - I assume our problems with this field come from the fact that we mix serialization and inheritance (here in the client and in the agent)